omc3/scripts/kmod_summary.py

```diff
@@ -76,144 +99,235 @@ def output_kmod_summary_tables(
-    for beam in beams:
-        LOG.debug(f"Processing result for: {beam}")
-        if grouped[beam]:
-            kmod_summary = tfs.concat(grouped[beam], ignore_index=True)
-            kmod_summary_x = kmod_summary[["IP", "NAME"] + cols_x]
-            kmod_summary_y = kmod_summary[["IP", "NAME"] + cols_y]
-            if averaged_meas:
-                kmod_summary_averaged = tfs.concat(grouped_averaged[beam], ignore_index=True)
-                kmod_summary_x_averaged = kmod_summary_averaged[["IP"] + cols_x]
-                kmod_summary_y_averaged = kmod_summary_averaged[["IP"] + cols_y]
-            table_logbook = []
-            for plane, df in [("X", kmod_summary_x), ("Y", kmod_summary_y)]:
-                table_str = df.to_string()
-                table_logbook.append(f"\n============================================ {beam} Results ({plane}-plane) ====================================================\n\n{table_str}\n")
-            if averaged_meas:
-                for plane, df_averaged in [("X", kmod_summary_x_averaged), ("Y", kmod_summary_y_averaged)]:
-                    table_str_averaged = df_averaged.to_string()
-                    table_logbook.append(f"\n======================================== {beam} Averaged Results ({plane}-plane) ===============================================\n\n{table_str_averaged}\n")
-            kmod_summaries[beam] = kmod_summary
-            tables_logbook[beam] = table_logbook
+                second_magnet = label.split("-")[1] # takes right magnet from magnet names. ex. [MQXA1.R5] from [MQXA1.L5-MQXA1.R5]
+                relevant = second_magnet.split(".")[-1] # takes last part of the second magnet name. ex. [R5] from [MQXA1.R5]
+                ip_number = relevant[1:] # isolate IP number. ex. [5] from [R5]
+                ip_name = f"{IP_COLUMN}{ip_number}"
+            df = data_file[COLS_X + COLS_Y].iloc[[0]] # returns a DataFrame with one row
+            df.insert(0, NAME_COLUMN, meas_name)
+            df.insert(0, IP_COLUMN, ip_name)
+            grouped.append(df)
+    return grouped
+
+def _collect_averaged_kmod_results(
+    averaged_meas: dict[str, dict[int, tfs.TfsDataFrame]],
+    beam: int
+    ) -> list[tfs.TfsDataFrame]:
 
-    if output_dir is not None:
-        for beam in beams:
-            _save_outputs(beam=beam, save_output_dir=output_dir, table_logbook=tables_logbook[beam], summary_df=kmod_summaries[beam])
-            if logbook:
-                _logbook_entry(logbook=logbook, beam=beam, save_output_dir=output_dir)
-        if not logbook:
-            LOG.info("Logbook name not provided: logbook entry not created.")
-    else:
-        LOG.info("Output_dir not provided: skipping file output for all beams.")
+    """
+    Gathers the various averaged Kmod results dataframes, taking only cols_x and cols_y values.
+
+    Args:
+        meas_paths (Sequence[Path | str]): List of measurement directories containing beam subfolders.
```

Comment:
This is not the name of the parameter.
It's also not the correct type of the parameter.
Please update.